File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/tags/ColorContentTag.java
package com.mcreater.fxfluent.xaml.tags;

import javafx.scene.paint.Color;

import static com.mcreater.fxfluent.util.ColorUtil.parseXamlColor;

public class ColorContentTag extends SimpleContentTag<Color> {
    public Color toObject() {
        String content = element.getData().toString();

        if (!element.elements().isEmpty()) return null;
        else return parseXamlColor(content);
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/xaml/tags/ColorContentTag.kt
package com.mcreater.fxfluent.xaml.tags

import com.mcreater.fxfluent.util.ColorUtil.Companion.parseXamlColor
import javafx.scene.paint.Color


open class ColorContentTag : SimpleContentTag<Color>() {
    override fun toObject(): Color {
        val content = element?.data.toString()
        return if (element?.elements()?.isNotEmpty()!!) Color.TRANSPARENT else parseXamlColor(content)
    }
}

